components/Activitylist.js

```diff
@@ -40,6 +49,33 @@ export default function Activitylist({
     }
   }
 
+  function handleAddFavoriteButton(id) {
+    const activitySuggestionCard = activityCards.find((card) => card.id === id);
+
+    if (activitySuggestionCard) {
+      setActivityCards(
+        activityCards.map((card) =>
+          card.id === id
+            ? {
+                ...card,
+                userData: card.userData.some((user) => user.userID === userID)
+                  ? card.userData.filter((user) => user.userID !== userID)
+                  : [
+                      ...card.userData,
+                      {
+                        userID: userID,
+                        userName: userName,
+                      },
+                    ],
+              }
+            : card
+        )
+      );
```

Comment:
this code is super hard to read. It's not a shame to use explicit return statements and one or two variables. You will be greatful for that in 2 weeks ;)